### ISSUE TITLE
Add ability to specify context to `winget upgrade`

### ISSUE THREAD

**Issue by o-l-a-v** [microsoft/winget-cli#2145]
### Description of the new feature / enhancement

Add a switch for context with ```winget upgrade```, for for listing upgrades (specifying no apps), and when upgrading a specific app.

#### List upgradable apps

* A user without admin permissions wan't to check what apps one can upgrade.
  * ```winget upgrade --context user```
* Running winget in system context, then upgrading user context apps makes no sense.
  * ```winget upgrade --context system```

#### Upgrade app

* Teams is installed with the system wide MSI installer, but user without admin permissions wants to upgrade Teams in system context.
  * ```winget upgrade --exact --id Microsoft.Teams --context user```

**Comment by cjwijtmans** [comment]
the existing flag would be --scope not --context. Not sure if consistency would be prefer or avoiding confusion would be prefer. 🤷🏼‍♂️

**Comment by o-l-a-v** [comment]
I agree, `--scope` for consistency.

**Comment by eabase** [comment]
Also very useful if the scope is shown when listing.

I.e. provide the functionality for added column with _scope_. 
```powershell
# Adding scope column in output
winget list --scope

# only showing items with machine scope
winget list --scope machine
```

**Comment by Trenly** [comment]
[Policy] Command-Upgrade

**Comment by eabase** [comment]
@Trenly

Sorry, I don't understand your answer/comment.

**Comment by Trenly** [comment]
> @Trenly
> 
> Sorry, I don't understand your answer/comment.

I was simply applying a label

**Comment by github-account1111** [comment]
> Teams is installed with the system wide MSI installer, but user without admin permissions wants to upgrade Teams in user context.
> 
> * `winget upgrade --exact --id Microsoft.Teams --scope user`

What is supposed to happen in this scenario? Should winget install the user-wide version of Teams (alongside the outdated system-wide one)?

What's the current behavior when a program is installed in both scopes, and `winget upgrade --id <AppID> is run?

**Comment by o-l-a-v** [comment]
Teams system wide installer is, at least used to be, a special one. It installs Teams in user context for all users that log in on that device and don't have Teams installed already. Thus in this scenario I'm pretty sure using WinGet to update Teams in user context would woek just fine.

**Comment by github-account1111** [comment]
Okay, you should've probably specified that in the issue because that's definitely not common knowledge or how most programs behave. Most programs, when installed with a system-wide installer, are installed system-wide, i.e. in *Program Files*, not just for all users.

So my question still stands.
